packages/workbox-core/src/types.ts

```diff
@@ -48,12 +48,23 @@ export interface RouteMatchCallback {
  * Options passed to a `RouteHandlerCallback` function.
  */
 export interface RouteHandlerCallbackOptions {
-  request: Request | string;
-  url?: URL;
+  event: ExtendableEvent;
+  request: Request;
+  url: URL;
   params?: string[] | MapLikeObject;
+}
+
+/**
+ * Options passed to a `ManualHandlerCallback` function.
+ */
+export interface ManualHandlerCallbackOptions {
+  request: Request | string;
   event?: ExtendableEvent;
 }
 
+export type HandlerCallbackOptions =
```

Comment:
Similar question to above. Would it work to make something like `HandlerCallbackOptions` a base type, and then any modules that need to accept `RouteHandlerCallbackOptions | ManualHandlerCallbackOptions` could accept any subtype of `HandlerCallbackOptions`?

Reply:
See my earlier comment about wanting to keep `RouteHandlerCallbackOptions` and `ManualHandlerCallbackOptions` independent. Making `HandlerCallbackOptions` a composite of both of them seems to work for our use case.

Reply:
Kk